### PR TITLE
Fix type filter for media content type

### LINKED ISSUE #2762: Media selection types filter not working
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | dev-develop (22.07.2016)

#### Actual Behavior

Type filter for media selection not working. 

#### Expected Behavior

Media selection types filter should work.

#### Steps to Reproduce

Try example.xml or create one with a document filter

```xml
<property name="documents" type="media_selection">
    <meta>
        <title lang="de">Dokumente</title>
        <title lang="en">Documents</title>
    </meta>
    <params>
        <param name="types" value="document"/>
        </params>
</property>
```

Open the template select medias and also images are shown.

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2762
| License | MIT

#### What's in this PR?

Fixed that media are not filter in the media overlay by configured type.

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js [hunk 1]
```diff
@@ -275,12 +280,23 @@ define([
             this.$el.find('.list-title').text(title);
         },
 
+        getUrl: function(params) {
+            if (params) {
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js]: I guess that should be `!params`.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js [hunk 2]
```diff
@@ -75,7 +76,11 @@ define([
 
             templates: {
                 skeleton: skeletonTemplate,
-                uploadUrl: '/admin/api/media?collection=<%= id %>&locale=<%= locale %><% if (!!types) {%>&types=<%= types %><% } %>'
+                url: [
+                    '<%= url %>?locale=<%= locale %>',
+                    '<% if (!!types) {%>&types=<%= types %><% } %>',
+                    '<% _.each(params, function(value, key) {%>&key=<%= value %><% }) %>'
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js]: Doesn't params also contain `types`? Or do you need both previous lines?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js [hunk 3]
```diff
@@ -275,12 +280,23 @@ define([
...
+            });
+        },
+
         changeUploadCollection: function(id) {
             this.sandbox.emit(
-                'husky.dropzone.' + this.options.instanceName + '.change-url', this.templates.uploadUrl({
-                    id: id,
-                    locale: this.options.locale,
-                    types: this.options.types
+                'husky.dropzone.' + this.options.instanceName + '.change-url', this.getUrl({
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js]: That's not resulting in the same URL, is it? Did the previous one not work?

Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js]: fixed

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js (hunk 2 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js]: This line actually creates the key `key`, which makes the upload in the overlay stop working.

**Pull request merged**